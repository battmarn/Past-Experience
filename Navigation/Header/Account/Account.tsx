import type { User } from '@trinnylondon/global.types.auth';
import type { SiteString } from '@trinnylondon/global.types.config';
import cx from 'classnames';
import React, { type FC } from 'react';

import gs from 'app/assets/css/global/globalStyles';
import AcccountIcon from 'app/assets/icons/my-account.icon.svg';
import { IconChevron } from 'app/bit-migration/react-components/atomics/atoms/icon-chevron/icon-chevron';
import { Link } from 'app/bit-migration/react-components/link/link';
import { useNavigation } from 'app/bit-migration/react-contexts/navigation';
import { Icon } from 'app/components/Icon/icon';
import { useSignOut } from 'app/hooks/auth/signOut';
import { useGtmSiteInteraction } from 'app/hooks/gtm/useGtmSiteInteraction';
import { useSnowplowSiteInteraction } from 'app/hooks/snowplow/useSnowplowSiteInteraction';
import { useSiteWideText } from 'app/hooks/useRegionalData';
import { useScrollLock } from 'app/hooks/useScrollLocker/useScrollLocker';
import { useSiteId } from 'app/hooks/useSiteId';

import { useNavigationContext } from '../../Context/Context';

import s from './styles.module.css';

export interface AccountProps {
  currentUser?: User;
}

const shouldShowDropdown = () => {
  return (
    window &&
    window.navigator &&
    !/android|webos|iphone|ipod|ipad|blackberry|iemobile|opera mini/i.test(window.navigator.userAgent)
  );
};

const accountDropdownSymbol = Symbol('ACCOUNT_DROPDOWN');

export const Account: FC<AccountProps> = ({ currentUser }) => {
  const {
    menu: { active: activeMenuItem, hideMenu, showMenu },
  } = useNavigationContext();

  const { signOut } = useSignOut();
  const { push } = useNavigation();
  const gtmSiteInteraction = useGtmSiteInteraction();
  const snowplowSiteInteraction = useSnowplowSiteInteraction();

  const hasCurrentUser = Boolean(currentUser && currentUser.id);

  const isAccountDropdownOpen = activeMenuItem === accountDropdownSymbol && hasCurrentUser;

  useScrollLock(accountDropdownSymbol, isAccountDropdownOpen);

  // The SiteString type is case sensitive...
  const currentSiteId = useSiteId().toLocaleUpperCase() as SiteString;
  const helpLinkSuffix = currentSiteId === 'DE' ? 'de' : `en-${currentSiteId.toLocaleLowerCase()}`;

  const showAccountDropdown = () => {
    if (hasCurrentUser && shouldShowDropdown()) {
      showMenu(accountDropdownSymbol);
      if (!isAccountDropdownOpen) {
        gtmSiteInteraction('Opened Account Dropdown', 'icon click');
        snowplowSiteInteraction('Opened account dropdown');
      }
    }
  };

  const hideAccountDropdown = (immediate: boolean = true) => {
    hideMenu(accountDropdownSymbol, immediate);
  };

  const onAccountClick = () => {
    if (!hasCurrentUser) {
      push('/sign-in');
      gtmSiteInteraction('Clicked - Sign In', 'icon click');
      snowplowSiteInteraction('Clicked - Sign in');
    } else {
      showAccountDropdown();
    }
  };

  const onLinkClick = (linkName: string) => {
    gtmSiteInteraction(`Clicked - ${linkName}`, 'click', 'Account Dropdown');
    snowplowSiteInteraction(`Clicked - ${linkName}`, 'Account dropdown');
  };

  const handleSignOut = () => {
    gtmSiteInteraction('Clicked - Sign Out', 'click', 'Account Dropdown');
    snowplowSiteInteraction('Clicked - Sign out', 'Account dropdown');
    void signOut();
    hideAccountDropdown(true);
  };

  return (
    <div className={s.account}>
      <button
        className={gs.accessibleFocus}
        aria-expanded={isAccountDropdownOpen}
        onMouseEnter={() => showAccountDropdown()}
        onMouseLeave={() => hideAccountDropdown(false)}
        onClick={onAccountClick}
        data-test-id="navigation-account"
      >
        <Icon className={s.accountIcon} aria-label="My Account">
          <AcccountIcon />
        </Icon>
      </button>
      {/* eslint-disable-next-line jsx-a11y/interactive-supports-focus -- We should look to fix this site wide for more than a11y but ensuring we don't have multiple menus open at once */}
      <div
        role="menu"
        hidden={!isAccountDropdownOpen}
        data-test-id="navigation-account-dropdown"
        className={cx(s.accountHoverDropdown, {
          [s.isInvisible]: !isAccountDropdownOpen,
        })}
        onMouseEnter={() => showAccountDropdown()}
        onMouseLeave={() => hideAccountDropdown()}
      >
        <span data-test-id="navigation-account-dropdown-name" className={s.accountName}>
          {}
          {`${useSiteWideText('hello', { defaultValue: 'hello' })}, ${currentUser?.firstName}`}
        </span>
        <ul className={s.accountLinksList} data-test-id="account-dropdown-links">
          <li className={s.accountLinkItem}>
            <Link
              className={s.accountLinkButton}
              data-test-id="account-link-profile"
              onClick={() => {
                onLinkClick('My Account');
                hideAccountDropdown(true);
              }}
              href={'/my-account'}
            >
              <span>{useSiteWideText('my-account', { defaultValue: 'My account' })}</span>
              <IconChevron style={{ width: '11px', height: '11px' }} direction="right" />
            </Link>
          </li>
          <li className={s.accountLinkItem}>
            <Link
              className={s.accountLinkButton}
              data-test-id="account-link-help"
              onClick={() => {
                onLinkClick('Help');
                hideAccountDropdown(true);
              }}
              href={`https://trinny-london.zendesk.com/hc/${helpLinkSuffix}`}
            >
              <span>{useSiteWideText('help', { defaultValue: 'Help' })}</span>
              <IconChevron style={{ width: '11px', height: '11px' }} direction="right" />
            </Link>
          </li>
          <li className={s.accountLinkItem}>
            <button className={s.accountLinkButton} data-test-id="account-link-signout" onClick={handleSignOut}>
              <span>{useSiteWideText('sign-out', { defaultValue: 'Sign out' })}</span>
              <IconChevron style={{ width: '11px', height: '11px' }} direction="right" />
            </button>
          </li>
        </ul>
      </div>
    </div>
  );
};
