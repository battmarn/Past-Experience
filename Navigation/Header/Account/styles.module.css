.account {
  position: relative;
  margin-right: 10px;
  text-decoration: none;
  display: none;

  @media (--desktop-menu) {
    display: grid;
    place-items: center;
  }
}

.accountIcon {
  color: var(--brand-grey-1);
  width: 24px;
  height: 24px;
  margin: 0;
  padding: 0;

  @media (--desktop-menu) {
    margin-top: 0;
  }
}

.accountHoverDropdown {
  color: var(--grey-600);
  background-color: var(--white);
  height: auto;
  position: absolute;
  top: 100%;
  right: 0;
  width: 350px;
  padding: 20px;
  box-shadow: var(--box-shadow);
  border: 1px solid var(--grey-500);
  z-index: var(--navigation-desktop-hover-z-index);
  overflow-y: auto;
  -webkit-overflow-scrolling: touch;
  max-height: calc(85vh - (var(--total-header-height) + 10px));
}

.accountHoverDropdown.isInvisible {
  display: none;
}

.accountName {
  font-size: var(--font-size-medium);
  font-weight: 800;
  text-transform: uppercase;
}

.accountLinksList {
  margin: 0;
  margin-top: var(--navigation-header-height);
  display: flex;
  flex-direction: column;
  list-style: none;
}

.accountLinkItem {
  margin: 8px 0px;
  background-color: var(--grey-100);
}

.accountLinkButton {
  text-decoration: none;
  font-weight: 800;
  display: flex;
  justify-content: space-between;
  padding: 10px 15px;
  align-items: center;
  width: 100%;
}

.rightIcon {
  width: 12px;
  height: 12px;
  color: var(--grey-600);
}
