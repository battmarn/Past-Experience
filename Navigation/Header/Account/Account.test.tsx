import { trackStructEvent } from '@snowplow/browser-tracker';
import type { User } from '@trinnylondon/global.types.auth';
import type { RegionString } from '@trinnylondon/global.types.config';
import React from 'react';
import { act } from 'react-dom/test-utils';

import { screen, userEvent, waitFor } from 'app/bit-migration/test-utilities/react-renderer';
import { createUseSignOutMock } from 'app/hooks/auth/signOut.mock';
import { createUseNavigationMock } from 'app/hooks/useNavigationMock';
import { createUseRegionalDataMock } from 'app/hooks/useRegionalDataMock';
import { createUseSiteIdMock } from 'app/hooks/useSiteIdMock';
import { clearDataLayer, snapshotDataLayer } from 'app/lib/analytics/analytics.mock';
import { createUseTrackingLocationMock } from 'app/providers/GTM/TrackingLocationProvider.mock';
import { mock } from 'test/helpers/mock';
import { renderWithStore } from 'test/helpers/renderWithStore';

import { NavigationContextProvider } from '../../Context/Context';

import { Account, type AccountProps } from './Account';

const user = userEvent.setup({ delay: null });

jest.mock('@snowplow/browser-tracker', () => ({
  trackStructEvent: jest.fn(),
}));

describe('<Account />', () => {
  let signOutMock: jest.Mock<Promise<void>>;

  const setup = (props: AccountProps, siteID: RegionString) => {
    createUseSiteIdMock(siteID);
    const { pushSpy } = createUseNavigationMock();
    const { container } = renderWithStore(
      <NavigationContextProvider>
        <Account {...props} />
      </NavigationContextProvider>,
    );
    return {
      container,
      pushSpy,
    };
  };

  beforeEach(() => {
    jest.useFakeTimers();
    clearDataLayer();
    createUseTrackingLocationMock('Header');
    signOutMock = createUseSignOutMock();
    createUseRegionalDataMock('siteWideText', {
      hello: 'Hello',
      'my-account': 'My account',
      help: 'Help',
      'sign-out': 'Sign out',
    });
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  describe('when currentUser is undefined', () => {
    it('will render account icon with correct href', () => {
      expect(setup({}, 'uk' as RegionString).container).toMatchSnapshot();
    });

    it('will track the click event', async () => {
      setup({}, 'uk' as RegionString);
      await user.click(screen.getByTestId('navigation-account'));
      snapshotDataLayer(1);

      expect(trackStructEvent).toHaveBeenCalledWith({
        category: 'Site interaction',
        action: 'Clicked - Sign in',
        label: 'Header',
      });
    });

    it('will push to the sign-in url on click', async () => {
      const { pushSpy } = setup({}, 'uk' as RegionString);
      await user.click(screen.getByTestId('navigation-account'));
      expect(pushSpy).toHaveBeenCalledWith('/sign-in');
    });
  });

  describe('when current user is passed in, but id is not present', () => {
    const propsWithFaultyUser = {
      currentUser: mock<User>({
        firstName: 'Skyler',
      }),
    };
    it('will render the SignUsers component with information to sign in', () => {
      expect(setup(propsWithFaultyUser, 'uk' as RegionString).container).toMatchSnapshot();
    });

    it('will track the click event', async () => {
      setup(propsWithFaultyUser, 'uk' as RegionString);
      await user.click(screen.getByTestId('navigation-account'));
      snapshotDataLayer(1);

      expect(trackStructEvent).toHaveBeenCalledWith({
        category: 'Site interaction',
        action: 'Clicked - Sign in',
        label: 'Header',
      });
    });

    it('will push to the sign-in url on click', async () => {
      const { pushSpy } = setup(propsWithFaultyUser, 'uk' as RegionString);
      await user.click(screen.getByTestId('navigation-account'));
      expect(pushSpy).toHaveBeenCalledWith('/sign-in');
    });
  });

  describe('when current user is valid', () => {
    const testPropsWithValidUser = {
      currentUser: mock<User>({
        firstName: 'Skyler',
        id: '1',
      }),
    };

    it('will render the account icon', () => {
      expect(setup(testPropsWithValidUser, 'uk' as RegionString).container).toMatchSnapshot();
    });

    it('will track the account icon click', async () => {
      setup(testPropsWithValidUser, 'uk' as RegionString);
      await user.click(screen.getByTestId('navigation-account'));
      snapshotDataLayer(1);

      expect(trackStructEvent).toHaveBeenCalledWith({
        category: 'Site interaction',
        action: 'Opened account dropdown',
        label: 'Header',
      });
    });

    it('will track the account icon hover', async () => {
      setup(testPropsWithValidUser, 'uk' as RegionString);
      await user.hover(screen.getByTestId('navigation-account'));
      snapshotDataLayer(1);
    });

    it('will hide the dropdown on unhover the icon after a timeout', async () => {
      setup(testPropsWithValidUser, 'uk' as RegionString);
      await user.hover(screen.getByTestId('navigation-account'));
      await waitFor(() => {
        expect(screen.getByTestId('navigation-account-dropdown')).toBeVisible();
      });
      await user.unhover(screen.getByTestId('navigation-account'));
      act(() => {
        jest.advanceTimersByTime(400);
      });
      await waitFor(() => {
        expect(screen.getByTestId('navigation-account-dropdown')).not.toBeVisible();
      });

      snapshotDataLayer(1);
    });

    it('will keep the dropdown visible if we unhover and hover the icon quickly', async () => {
      setup(testPropsWithValidUser, 'uk' as RegionString);
      await user.hover(screen.getByTestId('navigation-account'));
      await user.unhover(screen.getByTestId('navigation-account'));
      act(() => {
        jest.advanceTimersByTime(100);
      });
      await waitFor(() => {
        expect(screen.getByTestId('navigation-account-dropdown')).toBeVisible();
      });
      await user.hover(screen.getByTestId('navigation-account'));
      act(() => {
        jest.advanceTimersByTime(300);
      });
      await waitFor(() => {
        expect(screen.getByTestId('navigation-account-dropdown')).toBeVisible();
      });
      snapshotDataLayer(1);

      expect(trackStructEvent).toHaveBeenCalledWith({
        category: 'Site interaction',
        action: 'Opened account dropdown',
        label: 'Header',
      });
    });

    it('will keep the dropdown visible if we unhover the icon and hover the dropdown', async () => {
      setup(testPropsWithValidUser, 'uk' as RegionString);
      await user.hover(screen.getByTestId('navigation-account'));
      await user.unhover(screen.getByTestId('navigation-account'));
      act(() => {
        jest.advanceTimersByTime(100);
      });
      await waitFor(() => {
        expect(screen.getByTestId('navigation-account-dropdown')).toBeVisible();
      });
      await user.hover(screen.getByTestId('navigation-account-dropdown'));
      act(() => {
        jest.advanceTimersByTime(300);
      });
      await waitFor(() => {
        expect(screen.getByTestId('navigation-account-dropdown')).toBeVisible();
      });
      snapshotDataLayer(1);

      expect(trackStructEvent).toHaveBeenCalledWith({
        category: 'Site interaction',
        action: 'Opened account dropdown',
        label: 'Header',
      });
    });

    it('will hide the dropdown straight away after unhovering the dropdown', async () => {
      setup(testPropsWithValidUser, 'uk' as RegionString);
      await user.hover(screen.getByTestId('navigation-account'));
      await user.hover(screen.getByTestId('navigation-account-dropdown'));
      await user.unhover(screen.getByTestId('navigation-account-dropdown'));
      expect(screen.getByTestId('navigation-account-dropdown')).not.toBeVisible();
      snapshotDataLayer(1);

      expect(trackStructEvent).toHaveBeenCalledWith({
        category: 'Site interaction',
        action: 'Opened account dropdown',
        label: 'Header',
      });
    });

    it('will render the dropdown on click', async () => {
      setup(testPropsWithValidUser, 'uk' as RegionString);
      await user.click(screen.getByTestId('navigation-account'));
      expect(screen.getByTestId('navigation-account-dropdown')).toBeVisible();
    });

    it('will track an account dropdown item link click', async () => {
      setup(testPropsWithValidUser, 'uk' as RegionString);
      await user.click(screen.getByTestId('navigation-account'));
      await user.click(screen.getByTestId('account-link-help'));
      snapshotDataLayer(2);

      expect(trackStructEvent).toHaveBeenCalledWith({
        category: 'Site interaction',
        action: 'Opened account dropdown',
        label: 'Header',
      });

      expect(trackStructEvent).toHaveBeenCalledWith({
        category: 'Site interaction',
        action: 'Clicked - Help',
        label: 'Header - Account dropdown',
      });
    });

    it('will track the correct path for profile dropdown link when my account flag is on', async () => {
      setup(testPropsWithValidUser, 'uk' as RegionString);
      await user.click(screen.getByTestId('navigation-account'));
      await user.click(screen.getByTestId('account-link-profile'));
      snapshotDataLayer(2);

      expect(trackStructEvent).toHaveBeenCalledWith({
        category: 'Site interaction',
        action: 'Opened account dropdown',
        label: 'Header',
      });

      expect(trackStructEvent).toHaveBeenCalledWith({
        category: 'Site interaction',
        action: 'Clicked - My Account',
        label: 'Header - Account dropdown',
      });
    });

    it('will track the correct path for profile dropdown link when my account flag is off', async () => {
      setup(testPropsWithValidUser, 'uk' as RegionString);
      await user.click(screen.getByTestId('navigation-account'));
      await user.click(screen.getByTestId('account-link-profile'));
      snapshotDataLayer(2);

      expect(trackStructEvent).toHaveBeenCalledWith({
        category: 'Site interaction',
        action: 'Opened account dropdown',
        label: 'Header',
      });

      expect(trackStructEvent).toHaveBeenCalledWith({
        category: 'Site interaction',
        action: 'Clicked - My Account',
        label: 'Header - Account dropdown',
      });
    });

    it('will use the correct help link for a given siteID', async () => {
      const { container } = setup(testPropsWithValidUser, 'de' as RegionString);
      await user.click(screen.getByTestId('navigation-account'));
      expect(container).toMatchSnapshot();
    });

    it('will track the signout link on click', async () => {
      setup(testPropsWithValidUser, 'us' as RegionString);
      await user.click(screen.getByTestId('navigation-account'));
      await user.click(screen.getByTestId('account-link-signout'));
      snapshotDataLayer(2);

      expect(trackStructEvent).toHaveBeenCalledWith({
        category: 'Site interaction',
        action: 'Opened account dropdown',
        label: 'Header',
      });

      expect(trackStructEvent).toHaveBeenCalledWith({
        category: 'Site interaction',
        action: 'Clicked - Sign out',
        label: 'Header - Account dropdown',
      });
    });

    it('will sign out when the signout link is clicked', async () => {
      setup(testPropsWithValidUser, 'se' as RegionString);
      await user.click(screen.getByTestId('navigation-account'));
      await user.click(screen.getByTestId('account-link-signout'));
      expect(signOutMock).toHaveBeenCalled();
    });
  });
});
