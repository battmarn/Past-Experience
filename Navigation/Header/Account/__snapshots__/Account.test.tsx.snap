// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Account /> when current user is passed in, but id is not present will render the SignUsers component with information to sign in 1`] = `
<div>
  <div
    class="account"
  >
    <button
      aria-expanded="false"
      class="accessibleFocus"
      data-test-id="navigation-account"
    >
      <test-file-stub
        aria-label="My Account"
        classname="icon accountIcon"
      />
    </button>
    <div
      class="accountHoverDropdown isInvisible"
      data-test-id="navigation-account-dropdown"
      hidden=""
      role="menu"
    >
      <span
        class="accountName"
        data-test-id="navigation-account-dropdown-name"
      >
        Hello, Skyler
      </span>
      <ul
        class="accountLinksList"
        data-test-id="account-dropdown-links"
      >
        <li
          class="accountLinkItem"
        >
          <a
            class="accountLinkButton MuiBox-root emotion-0"
            data-test-id="account-link-profile"
            href="/my-account"
          >
            <span>
              My account
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
              data-test-id="icon-chevron-right"
              focusable="false"
              style="width: 11px; height: 11px;"
              viewBox="0 0 5 10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 1L5 5L1 9"
                stroke="currentColor"
                stroke-width="2"
              />
            </svg>
          </a>
        </li>
        <li
          class="accountLinkItem"
        >
          <a
            class="accountLinkButton MuiBox-root emotion-0"
            data-test-id="account-link-help"
            href="https://trinny-london.zendesk.com/hc/en-uk"
          >
            <span>
              Help
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
              data-test-id="icon-chevron-right"
              focusable="false"
              style="width: 11px; height: 11px;"
              viewBox="0 0 5 10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 1L5 5L1 9"
                stroke="currentColor"
                stroke-width="2"
              />
            </svg>
          </a>
        </li>
        <li
          class="accountLinkItem"
        >
          <button
            class="accountLinkButton"
            data-test-id="account-link-signout"
          >
            <span>
              Sign out
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
              data-test-id="icon-chevron-right"
              focusable="false"
              style="width: 11px; height: 11px;"
              viewBox="0 0 5 10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 1L5 5L1 9"
                stroke="currentColor"
                stroke-width="2"
              />
            </svg>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`<Account /> when current user is passed in, but id is not present will track the click event 1`] = `
Array [
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "icon click",
      "location": "Header",
      "name": "Clicked - Sign In",
    },
  },
]
`;

exports[`<Account /> when current user is valid will hide the dropdown on unhover the icon after a timeout 1`] = `
Array [
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "icon click",
      "location": "Header",
      "name": "Opened Account Dropdown",
    },
  },
]
`;

exports[`<Account /> when current user is valid will hide the dropdown straight away after unhovering the dropdown 1`] = `
Array [
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "icon click",
      "location": "Header",
      "name": "Opened Account Dropdown",
    },
  },
]
`;

exports[`<Account /> when current user is valid will keep the dropdown visible if we unhover and hover the icon quickly 1`] = `
Array [
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "icon click",
      "location": "Header",
      "name": "Opened Account Dropdown",
    },
  },
]
`;

exports[`<Account /> when current user is valid will keep the dropdown visible if we unhover the icon and hover the dropdown 1`] = `
Array [
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "icon click",
      "location": "Header",
      "name": "Opened Account Dropdown",
    },
  },
]
`;

exports[`<Account /> when current user is valid will render the account icon 1`] = `
<div>
  <div
    class="account"
  >
    <button
      aria-expanded="false"
      class="accessibleFocus"
      data-test-id="navigation-account"
    >
      <test-file-stub
        aria-label="My Account"
        classname="icon accountIcon"
      />
    </button>
    <div
      class="accountHoverDropdown isInvisible"
      data-test-id="navigation-account-dropdown"
      hidden=""
      role="menu"
    >
      <span
        class="accountName"
        data-test-id="navigation-account-dropdown-name"
      >
        Hello, Skyler
      </span>
      <ul
        class="accountLinksList"
        data-test-id="account-dropdown-links"
      >
        <li
          class="accountLinkItem"
        >
          <a
            class="accountLinkButton MuiBox-root emotion-0"
            data-test-id="account-link-profile"
            href="/my-account"
          >
            <span>
              My account
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
              data-test-id="icon-chevron-right"
              focusable="false"
              style="width: 11px; height: 11px;"
              viewBox="0 0 5 10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 1L5 5L1 9"
                stroke="currentColor"
                stroke-width="2"
              />
            </svg>
          </a>
        </li>
        <li
          class="accountLinkItem"
        >
          <a
            class="accountLinkButton MuiBox-root emotion-0"
            data-test-id="account-link-help"
            href="https://trinny-london.zendesk.com/hc/en-uk"
          >
            <span>
              Help
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
              data-test-id="icon-chevron-right"
              focusable="false"
              style="width: 11px; height: 11px;"
              viewBox="0 0 5 10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 1L5 5L1 9"
                stroke="currentColor"
                stroke-width="2"
              />
            </svg>
          </a>
        </li>
        <li
          class="accountLinkItem"
        >
          <button
            class="accountLinkButton"
            data-test-id="account-link-signout"
          >
            <span>
              Sign out
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
              data-test-id="icon-chevron-right"
              focusable="false"
              style="width: 11px; height: 11px;"
              viewBox="0 0 5 10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 1L5 5L1 9"
                stroke="currentColor"
                stroke-width="2"
              />
            </svg>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`<Account /> when current user is valid will track an account dropdown item link click 1`] = `
Array [
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "icon click",
      "location": "Header",
      "name": "Opened Account Dropdown",
    },
  },
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "click",
      "location": "Header - Account Dropdown",
      "name": "Clicked - Help",
    },
  },
]
`;

exports[`<Account /> when current user is valid will track the account icon click 1`] = `
Array [
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "icon click",
      "location": "Header",
      "name": "Opened Account Dropdown",
    },
  },
]
`;

exports[`<Account /> when current user is valid will track the account icon hover 1`] = `
Array [
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "icon click",
      "location": "Header",
      "name": "Opened Account Dropdown",
    },
  },
]
`;

exports[`<Account /> when current user is valid will track the correct path for profile dropdown link when my account flag is off 1`] = `
Array [
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "icon click",
      "location": "Header",
      "name": "Opened Account Dropdown",
    },
  },
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "click",
      "location": "Header - Account Dropdown",
      "name": "Clicked - My Account",
    },
  },
]
`;

exports[`<Account /> when current user is valid will track the correct path for profile dropdown link when my account flag is on 1`] = `
Array [
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "icon click",
      "location": "Header",
      "name": "Opened Account Dropdown",
    },
  },
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "click",
      "location": "Header - Account Dropdown",
      "name": "Clicked - My Account",
    },
  },
]
`;

exports[`<Account /> when current user is valid will track the signout link on click 1`] = `
Array [
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "icon click",
      "location": "Header",
      "name": "Opened Account Dropdown",
    },
  },
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "click",
      "location": "Header - Account Dropdown",
      "name": "Clicked - Sign Out",
    },
  },
]
`;

exports[`<Account /> when current user is valid will use the correct help link for a given siteID 1`] = `
<div>
  <div
    class="account"
  >
    <button
      aria-expanded="true"
      class="accessibleFocus"
      data-test-id="navigation-account"
    >
      <test-file-stub
        aria-label="My Account"
        classname="icon accountIcon"
      />
    </button>
    <div
      class="accountHoverDropdown"
      data-test-id="navigation-account-dropdown"
      role="menu"
    >
      <span
        class="accountName"
        data-test-id="navigation-account-dropdown-name"
      >
        Hello, Skyler
      </span>
      <ul
        class="accountLinksList"
        data-test-id="account-dropdown-links"
      >
        <li
          class="accountLinkItem"
        >
          <a
            class="accountLinkButton MuiBox-root emotion-0"
            data-test-id="account-link-profile"
            href="/my-account"
          >
            <span>
              My account
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
              data-test-id="icon-chevron-right"
              focusable="false"
              style="width: 11px; height: 11px;"
              viewBox="0 0 5 10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 1L5 5L1 9"
                stroke="currentColor"
                stroke-width="2"
              />
            </svg>
          </a>
        </li>
        <li
          class="accountLinkItem"
        >
          <a
            class="accountLinkButton MuiBox-root emotion-0"
            data-test-id="account-link-help"
            href="https://trinny-london.zendesk.com/hc/de"
          >
            <span>
              Help
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
              data-test-id="icon-chevron-right"
              focusable="false"
              style="width: 11px; height: 11px;"
              viewBox="0 0 5 10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 1L5 5L1 9"
                stroke="currentColor"
                stroke-width="2"
              />
            </svg>
          </a>
        </li>
        <li
          class="accountLinkItem"
        >
          <button
            class="accountLinkButton"
            data-test-id="account-link-signout"
          >
            <span>
              Sign out
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
              data-test-id="icon-chevron-right"
              focusable="false"
              style="width: 11px; height: 11px;"
              viewBox="0 0 5 10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 1L5 5L1 9"
                stroke="currentColor"
                stroke-width="2"
              />
            </svg>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`<Account /> when currentUser is undefined will render account icon with correct href 1`] = `
<div>
  <div
    class="account"
  >
    <button
      aria-expanded="false"
      class="accessibleFocus"
      data-test-id="navigation-account"
    >
      <test-file-stub
        aria-label="My Account"
        classname="icon accountIcon"
      />
    </button>
    <div
      class="accountHoverDropdown isInvisible"
      data-test-id="navigation-account-dropdown"
      hidden=""
      role="menu"
    >
      <span
        class="accountName"
        data-test-id="navigation-account-dropdown-name"
      >
        Hello, undefined
      </span>
      <ul
        class="accountLinksList"
        data-test-id="account-dropdown-links"
      >
        <li
          class="accountLinkItem"
        >
          <a
            class="accountLinkButton MuiBox-root emotion-0"
            data-test-id="account-link-profile"
            href="/my-account"
          >
            <span>
              My account
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
              data-test-id="icon-chevron-right"
              focusable="false"
              style="width: 11px; height: 11px;"
              viewBox="0 0 5 10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 1L5 5L1 9"
                stroke="currentColor"
                stroke-width="2"
              />
            </svg>
          </a>
        </li>
        <li
          class="accountLinkItem"
        >
          <a
            class="accountLinkButton MuiBox-root emotion-0"
            data-test-id="account-link-help"
            href="https://trinny-london.zendesk.com/hc/en-uk"
          >
            <span>
              Help
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
              data-test-id="icon-chevron-right"
              focusable="false"
              style="width: 11px; height: 11px;"
              viewBox="0 0 5 10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 1L5 5L1 9"
                stroke="currentColor"
                stroke-width="2"
              />
            </svg>
          </a>
        </li>
        <li
          class="accountLinkItem"
        >
          <button
            class="accountLinkButton"
            data-test-id="account-link-signout"
          >
            <span>
              Sign out
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
              data-test-id="icon-chevron-right"
              focusable="false"
              style="width: 11px; height: 11px;"
              viewBox="0 0 5 10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 1L5 5L1 9"
                stroke="currentColor"
                stroke-width="2"
              />
            </svg>
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`<Account /> when currentUser is undefined will track the click event 1`] = `
Array [
  Object {
    "event": "site_interaction",
    "site_interaction_info": Object {
      "featureType": "icon click",
      "location": "Header",
      "name": "Clicked - Sign In",
    },
  },
]
`;
